src/layout/graph_search/custom_graph_search.cpp

```diff
@@ -222,6 +222,9 @@ std::optional<RoutingRegion> do_graph_search_route_ancilla(
         Vertex curr = frontier.top(); frontier.pop();
         size_t distance_to_curr = *predecessor_map[curr].distance;
 
+        if constexpr (heuristic != Heuristic::None)
+            if(curr == slice_searcher.target_vertex()) break;
```

Comment:
Got to spend some time thinking hard on whether regular Djikstra can also stop here. In the general case, I think it's a bit more subtle, but for square lattices that might be the case.

When I do remove the constraint on the heuristic Djikstra's performance is very close to `A*`.